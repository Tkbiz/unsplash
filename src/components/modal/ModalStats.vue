<template>
  <v-col cols="12"
         class="py-0 stats-block">
    <v-row>
      <h3 class="stats-title">
        Photo's statistic:
      </h3>
      <div v-if="photoStats.downloads"
           class="col-12 stats-info">
        Download:
        {{ photoStats.downloads.total }}
      </div>
      <div v-if="photoStats.likes"
           class="col-12 stats-info">
        Likes:
        {{ photoStats.likes.total }}
      </div>
      <div v-if="photoStats.views"
           class="col-12 stats-info">
        View:
        {{ photoStats.views.total }}
      </div>
      <div v-if="tags"
           class="col-12 stats-info">
        Tags:
        <span v-for="tag in tags"
              :key="tag.title">
                {{ tag.title }}
              </span>
      </div>
    </v-row>
  </v-col>
</template>

<script>
export default {
  name: 'ModalStats',
  props: {
    photoStats: {
      type: Object,
      required: true
    },
    tags: {
      type: Array,
      required: true
    }
  }
}
</script>
