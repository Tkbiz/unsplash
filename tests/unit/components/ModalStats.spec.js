import {mount, createLocalVue} from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import ModalStats from '@/components/modal/ModalStats'

Vue.use(Vuetify)

const localVue = createLocalVue()
localVue.use(Vuex)

describe('ModalStats.vue', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
    wrapper = mount(ModalStats, {
      localVue,
      vuetify,
      propsData: {
        photoStats: {},
        tags: []
      }
    })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('renders props when passed', () => {
     const photoStats = {}
     const tags = []

    expect(wrapper.props()).toEqual({photoStats, tags})
  })

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
