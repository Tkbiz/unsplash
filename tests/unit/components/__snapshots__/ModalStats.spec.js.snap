// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalStats.vue should match snapshot 1`] = `
<div class="py-0 stats-block col col-12">
  <div class="row">
    <h3 class="stats-title">
      Photo's statistic:
    </h3>
    <!---->
    <!---->
    <!---->
    <div class="col-12 stats-info">
      Tags:
    </div>
  </div>
</div>
`;
